entity/user/user_test.go

```diff
@@ -0,0 +1,83 @@
+package user_test
+
+import (
+	"testing"
+	"user-api/entity/user"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const idTestUser string = "123"
+const nameTestUser string = "Name"
+const emailTestUser string = "Email@gmail.com"
+const ageTestUser int = 12
+
+var addressesIdsTestUser []string = []string{"1", "2"}
+
+const msgErrorInvalidParamUser string = "Param different from expected when build User"
+const msgErrorCreateInvalidUser string = "User created without requested params"
+
+var userTest user.User
+
+func init() {
+
+	userTest, _ = user.
+		NewBuilder().
+		Id(idTestUser).
+		Name(nameTestUser).
+		Email(emailTestUser).
+		Age(ageTestUser).
+		AddressesIds(addressesIdsTestUser).
+		Build()
+}
+
+func TestGetId(t *testing.T) {
```

Comment:
I don't think these tests for getters and setters are adding much value. I think it would be better to test these methods through the code that uses them.
In this class, I believe only the addAddressId and addressExists would need separate tests